Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -924,24 +953,43 @@ namespace Babylon
                 return info.Env().Global().Get(JS_CLASS_NAME).As<Napi::Function>().New({info[0]});
             }
 
+            static Napi::Object New(const Napi::Env env, Napi::Object napiTransform)
+            {
+                return env.Global().Get(JS_CLASS_NAME).As<Napi::Function>().New({napiTransform});
+            }
+
             XRReferenceSpace(const Napi::CallbackInfo& info)
                 : Napi::ObjectWrap<XRReferenceSpace>{info}
             {
-                if (info[0].IsString())
-                {
-                    // TODO: Actually support the different types of reference spaces.
-                    const auto referenceSpaceType = info[0].As<Napi::String>().Utf8Value();
-                    assert(referenceSpaceType == XRReferenceSpaceType::UNBOUNDED || referenceSpaceType == XRReferenceSpaceType::VIEWER);
-                }
-                else
+                if (info.Length() > 0)
                 {
-                    // TODO: Actually take the offset into account.
-                    auto* transform = XRRigidTransform::Unwrap(info[0].As<Napi::Object>());
-                    assert(transform != nullptr);
-                    (void)transform;
+                    if (info[0].IsString())
+                    {
+                        // TODO: Actually support the different types of reference spaces.
+                        const auto referenceSpaceType = info[0].As<Napi::String>().Utf8Value();
+                        assert(referenceSpaceType == XRReferenceSpaceType::UNBOUNDED ||
+                            referenceSpaceType == XRReferenceSpaceType::VIEWER);
+                    }
+                    else
+                    {
+                        /*auto* transform = XRRigidTransform::Unwrap(info[0].As<Napi::Object>());
+                        assert(transform != nullptr);*/
```

Comment:
?

Reply:
This caused a build break on as the transform variable is unused when compiling retail.  I will just remove the code.